test/gtest/uct/ib/test_ib_md.cc

```diff
@@ -222,4 +225,75 @@ UCS_TEST_P(test_ib_md, smkey_reg_atomic)
     ucs_mmap_free(buffer, size);
 }
 
+void
+test_ib_md::test_mkey_pack_mt_internal(unsigned access_mask, bool invalidate)
+{
+    constexpr size_t size = UCS_MBYTE;
+    unsigned pack_flags, dereg_flags;
+    void* buffer;
+    int ret;
+    uct_mem_h memh;
+
+    ret = ucs_posix_memalign(&buffer, size, size, "mkey_pack_mt");
+    ASSERT_EQ(0, ret) << "Allocation failed";
+
+    if ((access_mask & UCT_MD_MEM_ACCESS_REMOTE_ATOMIC) && is_bf_arm()) {
+        UCS_TEST_SKIP_R("FIXME: AMO reg key bug on BF device, skipping");
+        return;
+    }
+
+    if (!is_supported_pack_mem_flags(access_mask)) {
+        UCS_TEST_SKIP_R("mkey pack isn't supported");
+    }
+
+    if (invalidate) {
+        pack_flags  = UCT_MD_MKEY_PACK_FLAG_INVALIDATE_RMA;
+        dereg_flags = UCT_MD_MEM_DEREG_FLAG_INVALIDATE;
+    } else {
+        pack_flags = dereg_flags = 0;
+    }
+
+    ucs_status_t status = reg_mem(access_mask, buffer, size, &memh);
```

Comment:
could remove status and expect directly on function calls

Reply:
```suggestion
    ASSERT_UCS_OK(reg_mem(access_mask, buffer, size, &memh));
```

Do you mean something like that?

Reply:
yes

Reply:
Done.